File: php/src/Model/WorkflowRunSteps.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * The per-step view of a run - what a canvas UI paints.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowRunSteps
{
    public function __construct(
        public readonly string $runId,
        public readonly string $status,
        /** @var list<WorkflowRunStep> */
        public readonly array $steps,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            runId: (string) ($data['runId'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            steps: Json::parseEach($data, 'steps', [WorkflowRunStep::class, 'fromArray']),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkflowRunSteps.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * The per-step view of a run — what a canvas UI paints.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowRunSteps {
    private String runId;
    private String status;
    private List<WorkflowRunStep> steps;

    public String getRunId() { return runId; }
    public String getStatus() { return status; }
    public List<WorkflowRunStep> getSteps() { return steps; }
}
